[shardhrv] iP

shardhrv opened the pull request:
hi! my name 
# JEFF
> rotted, my brain has become - Yoda probably idk 

JEFF unclutters your brain from all the important tasks that you have to remember, so you can fill it with _useless memes_ instead. JEFF is
- fast
- reliable
- always at your side
- **ALWAYS** at your side
~~beware the chatbot, for it hides in your walls.~~

## JEFF features:
- [x] Manage your tasks
- [x] Manage your deadlines
- [x] Manage your life (***COMING SOON***)


To all my fellow Java Programmers, you can clone my [repo](https://github.com/shardhrv/ip) and run the bot for yourself by running the `main` function that looks like 
```
public static void main(String[] args) {
        new JEFF().run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

shardhrv committed:
Rename main class from Duke.java to JEFF.java

shardhrv committed:
Add echo support

shardhrv committed:
Add support for list functionality

shardhrv committed:
Add Task.java and update JEFF.java to have task tracking functionality

shardhrv committed:
Add ToDo task type and update JEFF to be compatible using it

shardhrv committed:
Add Deadline.java

shardhrv committed:
Add deadline functionality to JEFF. Also made a fix to Deadline formatting in toString

shardhrv committed:
Add Event.java task

shardhrv committed:
Add Event functionality to JEFF

shardhrv committed:
Update EXPECTED.TXT

shardhrv committed:
Add test cases for the chatbot

shardhrv committed:
Improve test cases and add exception handling

shardhrv committed:
Add delete functionality

shardhrv committed:
update README.md

shardhrv committed:
Add exceptions for corrupted file save data

shardhrv committed:
Add support to convert tasks to CSV format

shardhrv committed:
Stop tracking /data/ directory

shardhrv committed:
Add support to save tasks locally

shardhrv committed:
Merge branch 'Level-7'

shardhrv committed:
Add date and time support for events and deadlines

shardhrv committed:
Add formatted date printing support for Event and Deadline

shardhrv committed:
Add proper LocalDateTime support for tasks

shardhrv committed:
Add getter for due date

shardhrv committed:
Add date search functionality

shardhrv committed:
Merge branch 'Level-8'

shardhrv committed:
Create storage class

shardhrv committed:
Create Ui class

shardhrv committed:
Create Parser class

shardhrv committed:
Create TaskList class

shardhrv committed:
Create abstract Command class

shardhrv committed:
Refactor code by adding Ui functionality to Ui class

shardhrv committed:
Refactor code to improve Storage class functionalities

shardhrv committed:
Refactor JEFF.java completely to follow OOP

shardhrv committed:
Merge branch 'A-MoreOOP'

shardhrv committed:
Package all files

shardhrv committed:
Merge branch 'A-Packages'

shardhrv committed:
Merge branch 'add-gradle-support'

shardhrv committed:
Add gradle support to master

shardhrv committed:
Add test for gradle

shardhrv committed:
Merge branch 'A-JUnit'

shardhrv committed:
Add non-trivial comments for documentation

shardhrv committed:
Add checkstyle

shardhrv committed:
Update checkstyle

shardhrv committed:
Add Find Command

shardhrv committed:
Merge tag 'A-JavaDoc'

shardhrv committed:
Merge tag 'Level-9'

shardhrv committed:
Merge tag 'A-CodingStandard'

shardhrv committed:
Refactor code to remove unused code in JEFF.java

shardhrv committed:
Refactor codebase to comply with checkstyle standards

shardhrv committed:
Merge branch 'A-CheckStyle'

yooplo commented on src/main/java/jeff/Ui.java:
```diff
@@ -0,0 +1,61 @@
+
+    // UI Elements
+    private static final String LINE = "--------------------------------------------";
+    private static final String LOGO =
```
impressive logo!

yooplo commented on src/main/java/jeff/Ui.java:
```diff
@@ -0,0 +1,61 @@
+        System.out.println("What can I do for you?");
+    }
+
+    public void showLine() {
```
Javadocs could be added here to the public methods that you have.

yooplo reviewed (commented):
Looks good :)
Small updates suggested

RobotWizzard commented on src/main/java/jeff/Jeff.java:
```diff
@@ -0,0 +1,54 @@
+    private TaskList tasks;
+
+    /**
+     * Constructs a JEFF application with the specified file path for storage.
```
I think the header comment here can be more specific. Since the constructor doesn't actually accept a file path as parameter, maybe specifying the hardcoded location of the data file is more appropriate:
```
/**
 * Constructs a JEFF application that stores data in "./data/JEFF.txt"
 */
```

RobotWizzard commented on src/main/java/jeff/Storage.java:
```diff
@@ -0,0 +1,226 @@
+     * @param dirPath The path to the directory where tasks are stored.
+     * @param filePath The path to the file in dirPath where tasks are stored.
+     */
+    public Storage(String dirPath, String filePath) {
```
Not sure if I like the name "filePath" here as it might be confusing, since it's similar to "dirPath". Perhaps "filename" is more accurate?

RobotWizzard commented on src/main/java/jeff/Storage.java:
```diff
@@ -0,0 +1,226 @@
+    }
+
+    private void processLine(String line, ArrayList<Task> taskList) throws LineCorruptedException {
+        // Split the line by ","
```
Food for thought: What if the description of the task includes ","? 🤔

RobotWizzard reviewed (commented):
I really like how you leave comments everywhere in your code, and also how organized your file structure is. Nice! 👍

shardhrv committed:
Add image resources

shardhrv committed:
Add GUI to the chatbot and made code checkstyle compliant

shardhrv committed:
Merge tag 'Level-10'

shardhrv committed:
Remove UITest.java

Removed the UI tests as they are now implemented via JavaFX GUIs rather than the console

shardhrv committed:
Add assertion

Added an assertion to check that the UI does

not allow an empty message to be printed by the chatbot.

shardhrv committed:
Refactor code according to code quality

Storage class: make date formatter private

In accordance with OOP principles, the fields of the

storage class are now private with accompanying getters.

shardhrv committed:
Refactor code to use streams

Find Command and List Command use streams to improve

the readability of the code.

shardhrv committed:
Merge pull request #1 from shardhrv/A-Streams

Add stream usage

shardhrv committed:
Merge pull request #2 from shardhrv/A-CodeQuality

Refactor code according to code quality

shardhrv committed:
Update Assertion

shardhrv committed:
Merge pull request #3 from shardhrv/A-Assertions

Update Assertion

Solomon0519 commented:
Organization of files is neat and good!

shardhrv committed:
Create gradle.yml

shardhrv committed:
Merge pull request #4 from shardhrv/A-CI

Create gradle.yml

shardhrv committed:
Add CommandFactory

CommandFactory will takeover the responsibility of parser for the creation of commands

shardhrv committed:
Update parser class

Parser class uses new commandFactory in order to parse commands

shardhrv committed:
Add an Edit Command functionality

shardhrv committed:
Integrate edit command into GUI

Overhaul of edit command as it was not compatible with the initial command

factory. Edit command is also now works by changing the task to the new editted one.

shardhrv committed:
Merge pull request #5 from shardhrv/C-Update

C update

shardhrv committed:
Refactor PrintCommand.java to reduce nesting

Improve readability by reducing nesting in if-else statements and
including more guard clauses. Also changed found to isFound to fit
coding conventions.

shardhrv committed:
Fix UI not displaying Error messages properly

Remove the colour flags in the message sent to Ui.showMessage(message)
as that would improperly display the flag.

shardhrv committed:
Update Gui colours

Make them look more pleasing to the eye (in my humble opinion)

shardhrv committed:
Add support for coloured error messages

Error messages will now display in a red text to make it clearer to the
user when an error has occurred. Improves error visibility.

shardhrv committed:
Merge pull request #7 from shardhrv/A-BetterGui

A better gui

shardhrv committed:
Add personality to the bot, I have

Made it so that all the speech that Jeff replies with is akin to Yoda.
Do not question why this is so, for it is.

shardhrv committed:
Merge pull request #8 from shardhrv/A-Personality

Add personality to the bot, I have

shardhrv committed:
Update README.md

Changed names from legacy Duke.java to new and improved Jeff.java

shardhrv committed:
Add Window title and icon for the chatbot

Create a screenshot for display purposes to be uploaded to the website,
and thus needed title and icons for looking nicer.

shardhrv committed:
Update User Guide

Added Comprehensive command list for the reference of the user that is
straightforward and easy to understand, with examples on how to use
the commands as well.

shardhrv committed:
Merge pull request #9 from shardhrv/A-UserGuide

Update User Guide

shardhrv committed:
Update README.md

Display Screenshot of UI in README
